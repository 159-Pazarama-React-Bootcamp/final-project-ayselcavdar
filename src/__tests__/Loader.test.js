import render from 'react-test-renderer';
import Loader from '../components/Loader/Loader';

describe('<Loader /> rendering', () => {
  it('Animated loader svg should only one svg element', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.find('svg')).toHaveLength(1);
  });

  it('Animated loader svg should only 3 path element', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.find('path')).toHaveLength(3);
  });

  it('Loader snapshots check', () => {
    const comp = render.create(<Loader />);
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
