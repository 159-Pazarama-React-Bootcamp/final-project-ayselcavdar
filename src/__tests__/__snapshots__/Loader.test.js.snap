// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> rendering Loader render check 1`] = `
<svg
  viewBox="0 0 100 100"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M20 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin={0}
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M30 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.2s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M40 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.4s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
</svg>
`;

exports[`<Loader /> rendering Loader snapshots check 1`] = `
<svg
  viewBox="0 0 100 100"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M20 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin={0}
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M30 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.2s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M40 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.4s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
</svg>
`;

exports[`Alert render check 1`] = `
<svg
  viewBox="0 0 100 100"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M20 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin={0}
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M30 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.2s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
  <path
    d="M40 50h4v10h-4z"
    fill="#7772FE"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0.4s"
      dur="0.6s"
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 20; 0 0"
    />
  </path>
</svg>
`;
